inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -17,55 +17,241 @@
 import org.apache.wicket.markup.html.form.NumberTextField;
 import org.apache.wicket.model.CompoundPropertyModel;
 import org.apache.wicket.model.IModel;
 import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wicketstuff.event.annotation.OnEvent;
 
+import de.agilecoders.wicket.webjars.request.resource.WebjarsCssResourceReference;
+import de.agilecoders.wicket.webjars.request.resource.WebjarsJavaScriptResourceReference;
 import de.tudarmstadt.ukp.clarin.webanno.api.JCasProvider;
 import de.tudarmstadt.ukp.clarin.webanno.api.annotation.action.AnnotationActionHandler;
 import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.AnnotatorState;
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.rendering.event.RenderAnnotationsEvent;
 import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaAjaxButton;
 import de.tudarmstadt.ukp.clarin.webanno.support.lambda.LambdaModelAdapter;
 import de.tudarmstadt.ukp.clarin.webanno.ui.annotation.AnnotationPage;
 import de.tudarmstadt.ukp.clarin.webanno.ui.annotation.sidebar.AnnotationSidebar_ImplBase;
+import de.tudarmstadt.ukp.inception.log.EventRepository;
+import de.tudarmstadt.ukp.inception.log.model.LoggedEvent;
 import de.tudarmstadt.ukp.inception.recommendation.api.RecommendationService;
 import de.tudarmstadt.ukp.inception.recommendation.api.model.Preferences;
+import de.tudarmstadt.ukp.inception.recommendation.log.RecommenderEvaluationResultEventAdapter.Details;
+
+public class RecommendationSidebar extends AnnotationSidebar_ImplBase {
+	private static final long serialVersionUID = 4306746527837380863L;
+
+	private Logger log = LoggerFactory.getLogger(getClass());
+	private WebComponent chartContainer;
+
+	private @SpringBean RecommendationService recommendationService;
+	private @SpringBean EventRepository eventRepo;
+
+	IModel<AnnotatorState> aModel;
+
+	int maxPointsToPlot = 50;
+
+	public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel, AnnotationActionHandler aActionHandler,
+			JCasProvider aJCasProvider, AnnotationPage aAnnotationPage) {
+		super(aId, aModel, aActionHandler, aJCasProvider, aAnnotationPage);
+		this.aModel = aModel;
+		IModel<Preferences> model = LambdaModelAdapter.of(
+				() -> recommendationService.getPreferences(aModel.getObject().getUser(),
+						aModel.getObject().getProject()),
+				(v) -> recommendationService.setPreferences(aModel.getObject().getUser(),
+						aModel.getObject().getProject(), v));
+
+		Form<Preferences> form = new Form<>("form", CompoundPropertyModel.of(model));
+
+		form.add(new NumberTextField<Integer>("maxPredictions", Integer.class).setMinimum(1).setMaximum(10).setStep(1));
+
+		form.add(new CheckBox("showAllPredictions"));
+
+		form.add(new LambdaAjaxButton<>("save", (_target, _form) -> aAnnotationPage.actionRefreshDocument(_target)));
+
+		add(form);
+
+		chartContainer = new Label("chart-container");
+		chartContainer.setOutputMarkupId(true);
```

Comment:
It would be nice to extract wicket ids into `private static final String`, e.g. like we do it in `RecommenderEditorPanel`.

Reply:
Done